ecs/pod_creator.go

```diff
@@ -27,11 +31,235 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
 
 // CreatePod creates a new pod backed by AWS ECS.
 func (m *BasicECSPodCreator) CreatePod(ctx context.Context, opts ...*cocoa.ECSPodCreationOptions) (cocoa.ECSPod, error) {
-	return nil, errors.New("TODO: implement")
+
+	mergedPodCreationOpts := cocoa.MergeECSPodCreationOptions(opts...)
+	mergedPodExecutionOpts := cocoa.MergeECSPodExecutionOptions(mergedPodCreationOpts.ExecutionOpts)
+
+	if err := mergedPodCreationOpts.Validate(); err != nil {
+		return nil, errors.Wrap(err, "invalid pod creation options")
+	}
+
+	if err := mergedPodExecutionOpts.Validate(); err != nil {
+		return nil, errors.Wrap(err, "invalid pod execution options")
+	}
+
+	secrets := m.getSecrets(mergedPodCreationOpts)
+
+	err := m.createSecrets(ctx, secrets)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating secrets")
+	}
+
+	taskDefinition, err := m.exportPodCreationOptions(ctx, mergedPodCreationOpts, secrets)
+	if err != nil {
+		return nil, errors.Wrap(err, "translating the register task definition input to the correct type")
+	}
+
+	registerOut, err := m.client.RegisterTaskDefinition(ctx, taskDefinition)
+	if err != nil {
+		return nil, errors.Wrap(err, "registering task definition")
+	}
+
+	if registerOut.TaskDefinition == nil || registerOut.TaskDefinition.TaskDefinitionArn == nil {
+		return nil, errors.New("expected a task definition from ECS, but none was returned")
+	}
+
+	taskDef := cocoa.NewECSTaskDefinition().SetID(*registerOut.TaskDefinition.TaskDefinitionArn).SetOwned(true)
+
+	runTask := m.exportTaskExecution(mergedPodExecutionOpts, *taskDef)
+
+	runOut, err := m.client.RunTask(ctx, runTask)
+	if err != nil {
+		return nil, errors.Wrap(err, "running task")
+	}
+
+	if runOut.Failures != nil {
+		catcher := grip.NewBasicCatcher()
+		for _, failure := range runOut.Failures {
+			catcher.Add(errors.Errorf("task '%s': %s: %s\n", *failure.Arn, *failure.Detail, *failure.Reason))
+		}
+		return nil, errors.Wrap(catcher.Resolve(), "running task")
+	}
+
+	if len(runOut.Tasks) == 0 || runOut.Tasks[0].TaskArn == nil {
+		return nil, errors.New("expected a task to be running in ECS, but none was returned")
+	}
+
+	resources := cocoa.NewECSPodResources().
+		SetCluster(*mergedPodExecutionOpts.Cluster).
+		SetSecrets(translatePodSecrets(secrets)).
+		SetTaskDefinition(*taskDef).
+		SetTaskID(*runOut.Tasks[0].TaskArn)
```

Comment:
`utility.FromStringPtr`

Reply:
Since `runOut.Tasks[0].TaskArn` was checked for nil-ness in the if statement above this, would we still change it to use `utility` or keep it as *?

Reply:
It doesn't matter that much in this case, but it's nice to know that utility conversions will never panic.